tests/object_file_system_test.php

```diff
@@ -930,4 +930,72 @@ public function test_is_configured_alternative_file_system_class_is_not_set() {
         unset($CFG->alternative_file_system_class);
         $this->assertFalse($this->filesystem->is_configured());
     }
+
+    /**
+     * Test rewrite_pluginfile_urls() S3
+     */
+    public function test_rewrite_pluginfile_urls_s3_proper() {
+        global $CFG, $ME;
+
+        if (!$CFG->phpunit_objectfs_s3_integration_test_credentials) {
+            $this->markTestSkipped('No S3 test credentials in config file');
+        }
+
+        $script = '/pluginfile.php';
+        $farg = '/3854/tool_objectfs/content/1/test.pdf';
+        $ME = "$script$farg";
+        $bucket = $CFG->phpunit_objectfs_s3_integration_test_credentials['s3_bucket'];
+        $s3_region = $CFG->phpunit_objectfs_s3_integration_test_credentials['s3_region'];
+        $bucketkeyprefix = 'test/';
+        $presignedurl = "https://$bucket.s3.$s3_region.amazonaws.com/$bucketkeyprefix?param1=val1&param2=val2";
+
+        $this->filesystem = new test_file_system();
+        $externalclient = $this->filesystem->get_external_client();
+        $this->set_externalclient_config('enablepresignedurls', '1');
+        $this->set_externalclient_config('signingmethod', 's3');
+        $this->set_externalclient_config('bucketkeyprefix', $bucketkeyprefix);
+        $this->assertTrue($this->filesystem->presigned_url_configured());
+
+        $embedded = $this->filesystem->embed_origin_url($presignedurl);
+        $textformat = 'Fake test with pdf %s';
+        $originaltext = sprintf($textformat, $embedded);
+
+        $this->assertEquals(
+            sprintf($textformat, $CFG->wwwroot.$ME),
+            $this->filesystem->rewrite_pluginfile_urls($originaltext)
+        );
+    }
+
+    /**
+     * Test rewrite_pluginfile_urls() CloudFront
+     */
+    public function test_rewrite_pluginfile_urls_s3_cf() {
```

Comment:
can you please turn this into a test with a dataProvider so we can throw a bunch of chunks of text at it and assert them all as a batch